stdlib/public/TensorFlow/Dataset.swift

```diff
@@ -95,21 +95,19 @@ extension Dataset : Sequence {
   }
 }
 
-/// FIXME(SR-8684): @convention(tensorflow) types crash SILGen when their
-/// parameters or results have generic parameters. This is blocking generic
-/// higher-order dataset transformation functions.
-public extension Dataset where Element == Tensor<Float> {
+public extension Dataset {
+
+  // Note that this Dataset API implementation uses an experimental tracing
+  // feature, which is not robust and does not have great diagnostics yet.
+
   @inlinable @inline(__always)
   func map(
-    _ transform: @convention(tensorflow) (Element) -> Element
+    _ transform: @escaping (Element) -> Element
```

Comment:
I think Element can be allowed to vary now.

Reply:
Yeah, we can change it to

```swift
func map<OtherElement : TensorGroup>(_ transform: (Element) -> OtherElement) -> ...
```